packages/truffle/test/scenarios/commandRunner.js

```diff
@@ -74,19 +78,22 @@ module.exports = {
 
       child.stdout.on("data", data => {
         // accumulate buffer from chunks
-        if (!seenChildPrompt) {
-          outputBuffer += data;
-        }
+        outputBuffer += data;
 
-        // child process is ready for input when it displays the readyPrompt
-        if (!seenChildPrompt && outputBuffer.includes(readyPrompt)) {
-          seenChildPrompt = true;
-          inputCommands.forEach(command => {
-            child.stdin.write(command + EOL);
-          });
-          child.stdin.end();
-        }
+        if (readyPromptRex.test(outputBuffer)) {
+          // set outputBuffer to remaining segment after final match
+          const segments = outputBuffer.split(readyPromptRex);
+          outputBuffer = segments.pop();
```

Comment:
Truffle's test data won't fail on this, but as a general purpose solution this might not work. For example:

```
`this is a line
truffle(develop)>
this is a line
truffle(develop)>
this is a line
`
```

You'll skip one of the `truffle(develop)>` lines, whereas I think you previous version would have properly handled it.